spec/std/signal_spec.cr

```diff
@@ -24,5 +24,30 @@ describe "Signal" do
     Process.kill Signal::USR2, Process.pid
   end
 
+  it "CHLD.reset sets default Crystal child handler" do
+    Signal::CHLD.reset
+    child = Process.new("true", shell: true)
+    child.wait # doesn't block forever
+  end
+
+  it "CHLD.ignore sets default Crystal child handler" do
+    Signal::CHLD.ignore
+    child = Process.new("true", shell: true)
+    child.wait # doesn't block forever
+  end
+
+  it "CHLD.trap is called after default Crystal child handler" do
+    called = false
+    child = Process.new("true", shell: true)
+
+    Signal::CHLD.trap do
```

Comment:
Isn't this a race condition with the child exiting? Why not install this before starting the process?

Reply:
I need the `child.pid` for the `Process.exists?` expectation.

I'm surprised that this works, actually, maybe it's because the signal goes through the event loop? Maybe I should harden this and call a shell sleep, and kill it, so we're sure that ain't a race condition.

Reply:
Just `child = nil`, set the signal handler using `not_nil!` and then start `true` should work great.